pkgs/servers/matrix-appservice-discord/default.nix

```diff
@@ -11,20 +19,33 @@ let
 in mkYarnPackage rec {
   pname = "matrix-appservice-discord";
 
-  # when updating, run `./generate.sh <git release tag>`
-  version = "1.0.0";
+  # when changing version, make sure to update the local copy of `package.json`
+  version = "3.0.0";
 
   src = fetchFromGitHub {
-    owner = "Half-Shot";
+    owner = "matrix-org";
     repo = "matrix-appservice-discord";
     rev = "v${version}";
-    sha256 = "0pca4jxxl4b8irvb1bacsrzjg8m7frq9dnx1knnd2n6ia3f3x545";
+    sha256 = "sha256-tSH8UfjfubIQupSZ9T9SbbKnJcFfGDj4DLDkOpKtGMs=";
   };
 
   packageJSON = ./package.json;
-  yarnNix = ./yarn-dependencies.nix;
+  offlineCache = fetchYarnDeps {
+    yarnLock = "${src}/yarn.lock";
+    sha256 = "sha256-UPhibxSqadu69OfY0FheXxRWK7H2SU+6awzUp8kGo4I=";
+  };
 
   pkgConfig = {
+    "@matrix-org/matrix-sdk-crypto-nodejs" = {
+      # Building this native Rust dependency requires the `napi` tool from
+      # `napi-rs`, which isn't available in nixpkgs at the time of writing.
+      # Because appservice-discord does not really make use of this library
+      # anyway, we can snip it off to neutralise loading errors.
+      postInstall = ''
+        truncate -s0 index.js
+      '';
+    };
+
```

Comment:
All tests are passing with this, but I didn't try running the program on my
server. It would be nice if someone could try actually running the service with
this to confirm that this isn't breaking anything.

Reply:
Works fine in my deployment.